x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/plan/logical/UnresolvedRelation.java

```diff
@@ -89,11 +89,11 @@ public boolean equals(Object obj) {
         }
 
         UnresolvedRelation other = (UnresolvedRelation) obj;
-        return source().equals(other.source())
-            && table.equals(other.table)
+        return Objects.equals(source(), other.source())
```

Comment:
why does `source()` need to be checked here, but isn't in some of the other hashCode and equals functions?

Reply:
It's an inconsistency - the source isn't relevant. Likely for UnresolvedRelation was added since a table or index declaration is not common and thus, its source was considered relevant.
The change above was added to avoid a NPE caused by a null (vs an empty) location.